src/Microsoft.Data.SqlClient/tests/ManualTests/ProviderAgnostic/MultipleResultsTest/MultipleResultsTest.cs

```diff
@@ -2,366 +2,154 @@
-
-            // Run Test
-            RunTest();
-
-            Console.Out.Flush();
-            Console.Out.Dispose();
-
-            // Recover the standard output stream
-            StreamWriter standardOutput = new StreamWriter(Console.OpenStandardOutput());
-            standardOutput.AutoFlush = true;
-            Console.SetOut(standardOutput);
-
-            // Compare output file
-            var comparisonResult = FindDiffFromBaseline(baselinePath, outputPath);
-
-            if (string.IsNullOrEmpty(comparisonResult))
-            {
-                return true;
-            }
-
-            Console.WriteLine("Test Failed!");
-            Console.WriteLine("Please compare baseline : {0} with output :{1}", Path.GetFullPath(baselinePath), Path.GetFullPath(outputPath));
-            Console.WriteLine("Comparison Results : ");
-            Console.WriteLine(comparisonResult);
-            return false;
-        }
+            command.CommandText = s_sqlStatement;
 
-        private void PrintException(Type expected, Exception e, params string[] values)
-        {
-            try
-            {
-                Debug.Assert(e != null, "PrintException: null exception");
-
-                _globalBuilder.Length = 0;
-                _globalBuilder.Append(e.GetType().Name).Append(": ");
-
-                if (e is COMException)
-                {
-                    _globalBuilder.Append("0x").Append((((COMException)e).HResult).ToString("X8"));
-                    if (expected != e.GetType())
-                    {
-                        _globalBuilder.Append(": ").Append(e.ToString());
-                    }
-                }
-                else
-                {
-                    _globalBuilder.Append(e.Message);
-                }
-                AssemblyFilter(_globalBuilder);
-                Console.WriteLine(_globalBuilder.ToString());
-
-                if (expected != e.GetType())
-                {
-                    Console.WriteLine(e.StackTrace);
-                }
-                if (values != null)
-                {
-                    foreach (string value in values)
-                    {
-                        Console.WriteLine(value);
-                    }
-                }
-                if (e.InnerException != null)
-                {
-                    PrintException(e.InnerException.GetType(), e.InnerException);
-                }
-                Console.Out.Flush();
-            }
-            catch (Exception f)
-            {
-                Console.WriteLine(f);
-            }
+            // ExecuteNonQuery will drain every result set, info message and exception, collating these into a single exception.
+            SqlException exNonQuery = Assert.Throws<SqlException>(() => command.ExecuteNonQuery());
+            Assert.Equal($"{ResultSet2_Error}\n{ResultSet4_Error}\n{ResultSet6_Error}\n{ResultSet8_Error}\n{ResultSet10_Error}\n" +
```

Comment:
Three things:
* Just to make this more readable, could we store this as a variable before the assert?
* Can we use `Environment.NewLine` instead of replacing `\r\n` with `\n`
* When defining the variable, can we put each message/error on a new line?

combine it all, it'd look like:
```
var expectedMessage =
    $"{ResultSet2_Error}${Environment.NewLine}" +
    $"{ResultSet4_Error}${Environment.NewLine}" +
    ...
```

Reply:
I've factored it out into a separate variable, and adjusted it to use Environment.NewLine. Taking a second look at the code though, I'm not sure whether it's more readable to have the original idea of `$"{ResultSet2_Error}\n{...}"` or a `string.Join` call. What do you think?

Reply:
Hmmm - hadn't thought about a String.Join(). I'll take either as long as it's using `Environment.NewLine` :) Feel free to pick whichever approach looks best to you!

Reply:
Sounds good thanks - I'll leave it as it is now